climbing_stairs/climbing_stairs.py

```diff
@@ -3,7 +3,22 @@
 import sys
 
 def climbing_stairs(n, cache=None):
-  pass 
+  # base cases
+  if n < 0:
+    return 0
+
+  if n == 0:
+    return 1
+  
+  if not cache:
+    cache = {}
+
+  if n not in cache:
+    cache[n] = climbing_stairs(n - 1, cache) + climbing_stairs(n - 2, cache) + climbing_stairs(n - 3, cache)
+
+  return cache[n]
+
```

Comment:
As you said in your commit message, there's a bug here, the test just hangs. You'll want to add a check for the cache's existence and if the cache entry at `n` is greater than 0, which returns the cache entry at `n`.